noir-projects/noir-protocol-circuits/crates/types/src/abis/accumulated_data/private_accumulated_data_builder.nr

```diff
@@ -74,36 +76,88 @@ impl PrivateAccumulatedDataBuilder {
             encrypted_log_preimages_length: self.encrypted_log_preimages_length,
             unencrypted_log_preimages_length: self.unencrypted_log_preimages_length,
             public_data_update_requests: [PublicDataUpdateRequest::empty(); MAX_PUBLIC_DATA_UPDATE_REQUESTS_PER_TX],
-            gas_used: self.gas_used.add(self.non_revertible_gas_used)
+            gas_used
         }
     }
 
+    pub fn to_metered_gas_used(self) -> Gas {
+        let mut metered_bytes = 0;
+
+        // note_hash_gas
+        for i in 0..self.new_note_hashes.storage.len() {
+            if !is_empty(self.new_note_hashes.get_unchecked(i)) {
+                metered_bytes += DA_BYTES_PER_FIELD;
+            }
+        }
```

Comment:
Does this builder have empty items? I'd expect it to contain only non-empty, since it uses BoundedVecs (otherwise, we could well use fixed-length arrays). We should be able to remove the loop and emptiness check, and just check the BoundedVec's length.

Reply:
Yeah I think both this and the way we split/meter the non/revertible sets are candidates for significant optimization passes